### PR TITLE
iDeep: F.concat

### PR DESCRIPTION
**Pull request by niboshi**
iDeep support in F.concat (Based on #4276.)

### CODE REVIEW

**Comment by niboshi** [comment]
Rebased.

**Comment by kmaehashi** [comment]
Please check Jenkins failure.

**Comment by niboshi** [comment]
Fixed.

File: chainer/functions/array/concat.py [hunk 1]
```diff
@@ -40,9 +41,22 @@ def check_type_forward(self, in_types):
                 type_check.expect(in_types[0].shape[d] == in_types[i].shape[d])
 
     def forward(self, xs):
+        if (intel64.should_use_ideep('>=auto')
+                and intel64.inputs_all_ready(xs, (4,))):
+
```
Comment (hvy) [inline chainer/functions/array/concat.py]: A minor but should we remove this empty line?

**Comment by kmaehashi** [comment]
LGTM except for @hvy's comment!

File: chainer/functions/array/concat.py (hunk 1 above)
Comment (niboshi) [reply chainer/functions/array/concat.py]: Fixed (added comment instead)

**Comment by kmaehashi** [comment]
LGTM! (jenkins passed, waiting for appveyor)

**Pull request merged**